queue/queue.py

```diff
@@ -1,21 +1,53 @@
+class Node:
+  def __init__(self, value, nextNode=None):
+    self.value = value
+    self.next = nextNode
+
 class Queue:
   def __init__(self):
     self.size = 0
     # what data structure should we
     # use to store queue elements?
     # a list?
+    # a singly linked list?
+    self.head = None
+    self.tail = None
     self.storage = []
     self.length = 0
 
-  def enqueue(self, item):
+  '''
+  tail            head    .next
+  __      __      __      __
+  |  | -> |  | -> |  | -> |  |
+  --      --      --      --
+  '''
+  def enqueue(self, item): # add to head
```

Comment:
these look veeeery similar to your Linked List implementations. maybe try reusing them?